File: webmagic-scripts/src/main/resources/ruby/defines.rb
def xpath str
  $page.getHtml().xpath(str).toString()
end
def css str
  $page.getHtml().css(str).toString()
end
def urls str
  links = $page.getHtml().links().regex(str).all();
  $page.addTargetRequests(links);
end

File: webmagic-scripts/src/main/resources/js/defines.js
function $(str){
    return page.getHtml().$(str).toString();
}
function xpath(str){
    return page.getHtml().xpath(str).toString();
}
function urls(str){
    links = page.getHtml().links().regex(str).all();
    page.addTargetRequests(links);
}
